pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -713,13 +713,15 @@ func (ue *unsupportedFieldError) Error() string {
 
 // IsInvalidNameArg checks whether the name contains invalid characters
 func IsInvalidNameArg(name string) bool {
-	re := regexp.MustCompile(`[^a-zA-Z0-9\-]+`)
-	return re.MatchString(name)
+	startsWithNonAlphanumericRe := regexp.MustCompile(`^[^A-Za-z0-9]`)
+	containsUnsupportedCharacterRe := regexp.MustCompile(`[^A-Za-z0-9\-_]+`)
+
+	return startsWithNonAlphanumericRe.MatchString(name) || containsUnsupportedCharacterRe.MatchString(name)
 }
 
 // errInvalidName error when invalid characters for a name is provided
 func ErrInvalidName(name string) error {
-	return fmt.Errorf("validation for %s failed, name must satisfy regular expression pattern: [a-zA-Z][-a-zA-Z0-9]*", name)
+	return fmt.Errorf("validation for %s failed, name must satisfy regular expression pattern: [A-Za-z0-9][A-Za-z0-9\\-_]*", name)
```

Comment:
```suggestion
	return fmt.Errorf("validation for %s failed, name must satisfy regular expression pattern: [a-zA-Z][a-zA-Z0-9\\-_]*", name)
```

Reply:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-using-console-create-stack-parameters.html

Reply:
As it stands, eksctl has been accepting names starting with a number because of the regex problem we mentioned